src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -1320,6 +1320,13 @@ static void XMLCALL fmi3_parse_element_end(void* c, const char *elm) {
 
     if (context->skipElementCnt) {
         context->skipElementCnt--;
+        // Need to clear buffer
+        // else: non-parsed attributes of elements that failed to parse will be parsed with next element
```

Comment:
Taking the suggestion on the comment. 

As for renaming the functions, this is a bit odd, since there is already a `fmi3_xml_get_attr_str` function, and the difference to `fmi3_xml_parse_attr_as_string` are the return types, string(const char**) vs jm_vector(char)*.

So... the most accurate rename would be `fmi3_xml_get_attr_as_jm_vector_char`, which doesn't quite feel right. 

One could look into the question if one of these 2 functions is actually redundant, but I would rather take such cleanup as a separate issue.

Reply:
Suggestion: Okay good.

Renaming: Yes, let's take that another time. It would require some more thinking through. I mainly wanted to express that it's something we probably should do in the future.